src/audio/pipeline/pipeline-graph.c

```diff
@@ -143,14 +143,17 @@ struct pipeline *pipeline_new(uint32_t pipeline_id, uint32_t priority, uint32_t
 		return NULL;
 	}
 
+	memset(&posn, 0, sizeof(posn));
 	/* just for retrieving valid ipc_msg header */
 	ipc_build_stream_posn(&posn, SOF_IPC_STREAM_TRIG_XRUN, p->comp_id);
 
-	p->msg = ipc_msg_init(posn.rhdr.hdr.cmd, posn.rhdr.hdr.size);
-	if (!p->msg) {
-		pipe_err(p, "pipeline_new(): ipc_msg_init failed");
-		rfree(p);
-		return NULL;
+	if (posn.rhdr.hdr.size) {
+		p->msg = ipc_msg_init(posn.rhdr.hdr.cmd, posn.rhdr.hdr.size);
```

Comment:
we need to validate the size to a max size value (macro in the headers ?).

Reply:
MAX size ?  How much ? Even it is not extreme large but a large one, it will affect the following allocation and result to random allocation failure

Reply:
@mwasko @marcinszkudlinski @rfredzim can someone help here for max size ?

Reply:
@lgirdwood I think this patch is good as it is now. The max size check can be a useful addition (maybe inside ipc_msg_init() is better), but it doesn't solve this problem. We are basicly creating ipc_msg_init() from random garbled data. E.g. we could create a MAX-1 byte IPC message that is not used for anything and cause weird problems (memory running out). 

So I think  Rander's fix is correct, only if ipc_build_stream_posn() sets a non-zero header size, we should proceed to allocate "p->msg".